### DISCUSSION TITLE
Group class closest search

### DISCUSSION THREAD

**Discussion by mikedejong** [tailwindlabs/tailwindcss#8719] [Ideas]
**What version of Tailwind CSS are you using?**
For example: v^3.1.2

**What build tool (or framework if it abstracts the build tool) are you using?**
For example: svelte v^3.48.0, postcss v^8.4.14

**What version of Node.js are you using?**
v17.8.0

**What browser are you using?**
Chrome

**What operating system are you using?**
macOS

**Describe your issue**
In the small example here below you see I've 2 `group` classes and when you hover the link I want to make the text red, but it will also trigger this action is also triggert when you hover over the `div1` element because the div1 has the `group` class also. Is it possible to look to the closest group and not all the `group` classes in the DOM tree? 

```
<div class="div1 group" open>
	<div class="opacity-0 group-open:opacity-100">
        Show when open class is added
    </div>
    <a class="div2 group">
        <img src="#" alt="">
        <span class="group-hover:text-red"></span>
    </a>
</div>
```

**Comment by adamwathan** [comment]
Yeah would love if we could do something like this but I've put a lot of time into trying different things and I just don't think it's possible with CSS 😭

Here's a related discussion with other ideas for tackling the same problem: https://github.com/tailwindlabs/tailwindcss/discussions/7547

**Reply by mikedejong** [reply]
Thanks for your help Adam, I'm going to change my current setup with a different approach. The solution with #7547 with group names (for now) is also a great solution! 😄
